File: java/common/src/main/java/com/ais/middleware/common/events/prs/ProducerCrossReferenceRetrievedEvent.java
package com.ais.middleware.common.events.prs;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.OffsetDateTime;

/**
 * Published by customer-identity-service after retrieving the producer cross-reference.
 * Consumed by: AppraisalReceivedSaga (prs-appraisal-service).
 *
 * ⚠️ DEMO GAP: ProducerControlCode routing logic unknown — UA vs UST rules need PRS team input.
 * Currently using inspectionTypeCode (A/B/I/J) as a proxy for routing decision.
 *
 * Topic: prs.events.producer-crossref-retrieved
 */
public record ProducerCrossReferenceRetrievedEvent(
        String correlationId,
        String policyNumber,
        // ⚠️ DEMO GAP: Real field names from CustomerDB stored procedure unknown
        String producerCode,
        String producerControlCode,
        OffsetDateTime retrievedAt
) {
    @JsonCreator
    public ProducerCrossReferenceRetrievedEvent(
            @JsonProperty("correlationId") String correlationId,
            @JsonProperty("policyNumber") String policyNumber,
            @JsonProperty("producerCode") String producerCode,
            @JsonProperty("producerControlCode") String producerControlCode,
            @JsonProperty("retrievedAt") OffsetDateTime retrievedAt
    ) {
        this.correlationId = correlationId;
        this.policyNumber = policyNumber;
        this.producerCode = producerCode;
        this.producerControlCode = producerControlCode;
        this.retrievedAt = retrievedAt;
    }
}

File: dotnet/Middleware.Contracts/Events/ProducerCrossReferenceRetrievedEvent.cs
namespace Middleware.Contracts.Events;

// UC4 · Appraisal Documents — Customer domain publishes producer cross-reference result
public sealed class ProducerCrossReferenceRetrievedEvent
{
    public string AppraisalId { get; set; } = string.Empty;
    public string PolicyNumber { get; set; } = string.Empty;

    // ⚠️ DEMO GAP: ProducerCode format from real Customer DB unknown — REPLACE_ME_PRODUCER_CODE_FORMAT
    public string ProducerCode { get; set; } = string.Empty;

    // ⚠️ DEMO GAP: UWControlCode routing rules not confirmed with PRS team — REPLACE_ME_UW_CONTROL_CODE_RULES
    // Expected values: "UA" (Underwriting Associate) or "UST" (Underwriting Specialist Team)
    public string UWControlCode { get; set; } = string.Empty;

    public string CorrelationId { get; set; } = string.Empty;
    public DateTimeOffset RetrievedAt { get; set; }
}
